lib/simulation.rb

```diff
@@ -0,0 +1,91 @@
+# Simulation acts as an all-knowing entity that is setting up and initializing
+# the board. The Simulation class is interacting heavily with a board and cells
+# because it knows the rules where a Board shouldn't carry that information
+# The rules are simply just bounds of the Board being initialized, being
+# highest, lowest, most left, and most right points. A simulation should
+# be created everytime a User would like to run a new randomly generated
+# "run" of this situation, being that there will be a target cell and
+# active cells nearby it which can be detected through diagonal, horizontal,
+# and vertical checks
+
+class Simulation
+  attr_reader :board,
+              :active_cells_near_target_count
+
+  def initialize(row = 1, column = 1)
+    @board = Board.new
+    @target_row = row
+    @target_column = column
+    @active_cells_near_target_count = 0
+    @out_of_bounds_top = 0 
+    @out_of_bounds_left = 0 
+    @out_of_bounds_bottom = @board.cells.count - 1
+    @out_of_bounds_right = @board.cells[row].count - 1
```

Comment:
Prompt says to expect 3x3, but would like to prepare for scaling of a larger board!